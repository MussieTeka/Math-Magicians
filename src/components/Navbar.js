import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/components/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <div className="navbar-left">
      <NavLink to="/">
        <h1>Math Magicians</h1>
      </NavLink>
    </div>
    <ul className="navbar-right">
      <li>
        <NavLink to="/" activeclassname="active">
          Home
        </NavLink>
      </li>
      <li>
        <NavLink to="/calculator" activeclassname="active">
          Calculator
        </NavLink>
      </li>
      <li>
        <NavLink to="/quote" activeclassname="active">
          Quotes
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
