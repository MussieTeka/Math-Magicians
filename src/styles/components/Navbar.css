.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  color: #4d4d4d;
}

.navbar-left a {
  text-decoration: none;
  font-size: 2rem;
  font-weight: bold;
  color: #4d4d4d;
  letter-spacing: 0.1em;
}

.navbar-right {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.navbar-right li {
  margin-left: 1.5rem;
}

.navbar-right a {
  text-decoration: none;
  font-size: 1.2rem;
  font-weight: bold;
  letter-spacing: 0.05em;
  transition: all 0.3s ease-in-out;
  color: #4d4d4d;
}

.navbar-right a.active {
  color: #007bff;
}

@media screen and (max-width: 768px) {
  .navbar {
    flex-direction: column;
    align-items: flex-start;
  }

  .navbar h1 {
    font-size: 2rem;
  }

  .navbar-right {
    margin-top: 1rem;
  }

  .navbar-right li {
    margin-left: 1rem;
  }

  .navbar-right a {
    font-size: 1rem;
  }
}
