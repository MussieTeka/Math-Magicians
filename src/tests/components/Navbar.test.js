import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../../components/Navbar';

describe('navbar', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <Router>
        <div>
          <Navbar />
        </div>
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
