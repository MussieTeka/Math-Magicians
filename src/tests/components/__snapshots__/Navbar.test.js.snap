// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar should render correctly 1`] = `
<div>
  <nav
    className="navbar"
  >
    <div
      className="navbar-left"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <h1>
          Math Magicians
        </h1>
      </a>
    </div>
    <ul
      className="navbar-right"
    >
      <li>
        <a
          activeclassname="active"
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
